pkg/gadgets/snapshot/process/tracer/gadget.go

```diff
@@ -0,0 +1,69 @@
+// Copyright 2022-2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tracer
+
+import (
+	columnhelpers "github.com/inspektor-gadget/inspektor-gadget/internal/column-helpers"
+	gadgetregistry "github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-registry"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/snapshot/process/types"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+const (
+	ParamThreads = "threads"
+)
+
+type Gadget struct{}
+
+func (g *Gadget) Name() string {
+	return "process"
+}
+
+func (g *Gadget) Category() string {
+	return gadgets.CategorySnapshot
+}
+
+func (g *Gadget) Type() gadgets.GadgetType {
+	return gadgets.TypeOneShot
+}
+
+func (g *Gadget) Description() string {
+	return "Gather information about running processes"
+}
+
+func (g *Gadget) Params() params.Params {
+	return params.Params{
+		{
+			Key:          ParamThreads,
+			Title:        "Show all threads",
```

Comment:
```suggestion
			Title:        "Threads",
			Description:  "Show all threads",
```

Reply:
I'll change this, but I think the titles will cause some discussion later on anyways. They're not used in the CLI, but would be used in a graphical UI like a web-interface or this tview based example I showed, where it makes sense to show more than just the key as the title on a form.

Reply:
Do you think we can drop "Title" now and introduce it later on? (Just trying to make this PR easier to review)

Reply:
I'd prefer not remove it, if that's okay, as it's already present in some gadgets and I have code that builds upon it (that GUI thingy and the web bridge).

Reply:
That's fine.